### PR TITLE
doc: tf-m: update thread with support information

### PR DESCRIPTION
**Pull request by greg-fer**
Added information about Thread supporting TF-M.
Updated structure on some pages for clarity.
Reworked formatting of the include tfm text file.
KRKNWK-10091.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/ug_nrf5340.rst [hunk 1]
```diff
@@ -74,6 +60,30 @@ In Zephyr, the application core is divided into two different build targets:
...
    In |NCS| releases before v1.6.1, the build target ``nrf5340dk_nrf5340_cpuapp_ns`` was named ``nrf5340dk_nrf5340_cpuappns``.
 
+The |NCS| provides two alternatives for running applications from the non-secure area of the memory: Secure Partition Manager and Trusted Firmware-M.
+
+Secure Partition Manager (SPM)
+------------------------------
+
+The :ref:`secure_partition_manager` sample uses the SPU peripheral to configure security attributions for flash, SRAM, and peripherals.
+After the configuration setup is complete, the sample loads the application firmware from the non-secure domain.
+In addition, the SPM sample can provide access to secure services to the application firmware.
```
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:70]:
```suggestion
In addition, the SPM sample provides the application firmware with access to secure services.
```

File: doc/nrf/ug_nrf5340.rst [hunk 2]
```diff
@@ -74,6 +60,30 @@ In Zephyr, the application core is divided into two different build targets:
...
+
+The SPM sample is currently the default solution used by most |NCS| samples.
+This means that when you build your application for the non-secure domain, the :ref:`secure_partition_manager` sample is automatically included in the build.
+
+Trusted Firmware-M (TF-M)
+-------------------------
+
+Trusted Firmware-M provides a highly configurable set of software components to create a Trusted Execution Environment.
```
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:78]: Might want to dump the word "highly".

File: doc/nrf/ug_nrf5340.rst [hunk 3]
```diff
@@ -74,6 +60,30 @@ In Zephyr, the application core is divided into two different build targets:
...
+TF-M is a framework which will be extended for new functions and use cases beyond the scope of SPM.
```
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:79]:
```suggestion
TF-M is a framework that will be extended with new functions and use cases beyond the scope of SPM.
```

File: doc/nrf/ug_nrf5340.rst [hunk 4]
```diff
@@ -74,6 +60,30 @@ In Zephyr, the application core is divided into two different build targets:
...
+
+Support for TF-M in the |NCS| is currently experimental.
+If your application does not depend on the secure services developed in SPM and does not use them, then TF-M can replace SPM as the secure firmware component in your application.
```
Comment (peknis) [inline doc/nrf/ug_nrf5340.rst:82]:
```suggestion
If your application does not depend on the secure services developed in SPM and does not use them, TF-M can replace SPM as the secure firmware component in your application.
```

File: doc/nrf/ug_thread_configuring.rst [hunk 5]
```diff
@@ -232,6 +232,20 @@ Minimal Thread Device (MTD)
   By default, the MTD operates as Minimal End Device (MED).
   To make it operate as Sleepy End Device (SED), set :kconfig:`CONFIG_OPENTHREAD_MTD_SED`.
 
+.. _thread_ug_tfm_support:
+
+Trusted Firmware-M support options
+==================================
+
+You can configure your Thread application to run with Trusted Firmware-M by enabling the following Kconfig options:
```
Comment (peknis) [inline doc/nrf/ug_thread_configuring.rst:240]:
```suggestion
To configure your Thread application to run with Trusted Firmware-M, enable the following Kconfig options:
```

File: samples/openthread/coap_client/README.rst [hunk 6]
```diff
@@ -51,29 +49,34 @@ The following CoAP resources are accessed on the server side:
 This sample uses :ref:`Zephyr CoAP API<zephyr:coap_sock_interface>` for communication, which is the preferred API to use for new CoAP applications.
 For example usage of the native Thread CoAP API, see the :ref:`coap_server_sample` sample.
 
-.. _coap_client_sample_multi_ext:
+Device Firmware Upgrade extension
+=================================
 
-Multiprotocol Bluetooth LE extension
-====================================
+This optional extension can be used to perform an over-the-air Device Firmware Upgrade (DFU).
```
Comment (peknis) [inline samples/openthread/coap_client/README.rst:55]:
```suggestion
You can use this optional extension to perform an over-the-air Device Firmware Upgrade (DFU).
```

File: doc/nrf/ug_thread_configuring.rst [hunk 7]
```diff
@@ -232,6 +232,20 @@ Minimal Thread Device (MTD)
...
+
+* :kconfig:`CONFIG_BUILD_WITH_TFM`
+* :kconfig:`CONFIG_OPENTHREAD_CRYPTO_PSA`
+
+These options are enabled by default for the :ref:`application core <ug_nrf5340_intro_app_core>` of the ``nrf5340dk_nrf5340_cpuapp_ns`` build target, which is supported by most of the :ref:`openthread_samples` in the |NCS|.
```
Comment (peknis) [inline doc/nrf/ug_thread_configuring.rst:245]: If "which" refers to the build target, it should maybe be "that" instead (without the comma, of course). Or you could split the long sentence -> ...build target. It is supported by most of the...

**Review by canisLupus1313 — APPROVED** [review]

File: samples/openthread/coap_client/README.rst [hunk 8]
```diff
@@ -51,29 +49,34 @@ The following CoAP resources are accessed on the server side:
...
 
-This optional extension can be used to perform an over-the-air Device Firmware Upgrade (DFU).
-In this process, the device that hosts the new firmware image sends it to the CoAP Client device using `SMP over Bluetooth`_.
-The :ref:`MCUboot <mcuboot:mcuboot_wrapper>` bootloader solution is then used to replace the old firmware image with the new one.
+Multiprotocol Bluetooth LE extension
+====================================
 
-.. note::
-   The Device Firmware Upgrade feature is currently supported only on the nRF52840 DK.
+This optional extension can demonstrate the OpenThread stack and :ref:`nrfxlib:softdevice_controller` working concurrently.
```
Comment (peknis) [inline samples/openthread/coap_client/README.rst:72]: can demonstrate or demonstrates?

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by edmont — APPROVED** [review]

**Review by wiba-nordic — COMMENTED** [review]
A couple of nits.

File: doc/nrf/ug_thread_configuring.rst (hunk 5 above)
Comment (greg-fer) [reply doc/nrf/ug_thread_configuring.rst:240]: Also added "on the nRF5340 DK"...

File: doc/nrf/ug_nrf5340.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/ug_nrf5340.rst:82]: Used "from".

File: doc/nrf/ug_nrf5340.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/ug_nrf5340.rst:79]: `It is a framework for functions and use cases beyond the scope of SPM.` @wiba-nordic ?

**Review by peknis — APPROVED** [review]

**Review by wiba-nordic — APPROVED** [review]

File: doc/nrf/includes/tfm.txt [hunk 9]
```diff
@@ -1,2 +1,3 @@
-When built for nrf5340dk_nrf5340_cpuapp_ns and nrf9160dk_nrf9160_ns targets, the sample is configured to compile and run as a non-secure application.
-Therefore, it automatically includes :ref:`TF-M <ug_tfm>` that prepares the required peripherals to be available for the application.
+When built for the ``nrf5340dk_nrf5340_cpuapp_ns`` target, the sample is configured to compile and run as a non-secure application.
+Therefore, it automatically includes :ref:`Trusted Firmware-M <ug_tfm>` that prepares the required peripherals to be available for the application.
```
Comment (Damian-Nordic) [inline doc/nrf/includes/tfm.txt:2]:
```suggestion
Therefore, it automatically includes :ref:`Trusted Firmware-M <ug_tfm>` that prepares the required peripherals and secure services to be available for the application.
```

File: doc/nrf/ug_thread_configuring.rst [hunk 10]
```diff
@@ -232,6 +232,21 @@ Minimal Thread Device (MTD)
...
+
+Trusted Firmware-M support options
+==================================
+
+To configure your Thread application on the nRF5340 DK to run with Trusted Firmware-M, enable the following Kconfig options:
+
+* :kconfig:`CONFIG_BUILD_WITH_TFM`
+* :kconfig:`CONFIG_OPENTHREAD_CRYPTO_PSA`
+
+These options are enabled by default for the :ref:`application core <ug_nrf5340_intro_app_core>` of the ``nrf5340dk_nrf5340_cpuapp_ns`` build target.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_thread_configuring.rst:245]: They are enabled by default for this build target, but only in the case of OT samples included in NCS. Not sure if that's what you meant.

File: doc/nrf/ug_nrf5340.rst [hunk 11]
```diff
@@ -46,25 +51,6 @@ Use this core for tasks that require high performance and for application-level
 
 The M33 TrustZone divides the application MCU into secure and non-secure domains.
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/ug_nrf5340.rst:52]: Follow-up: https://projecttools.nordicsemi.no/jira/browse/NCSDK-13898

File: doc/nrf/ug_thread_configuring.rst (hunk 10 above)
Comment (greg-fer) [reply doc/nrf/ug_thread_configuring.rst]: Rephrased slightly.

**Comment by greg-fer** [comment]
@Damian-Nordic , please re-review.

**Review by Damian-Nordic — APPROVED** [review]

**Pull request merged**